src/AsmResolver.DotNet/Collections/ParameterCollection.cs

```diff
@@ -142,6 +155,13 @@ private void UpdateParameterTypes()
             return _owner.ParameterDefinitions.FirstOrDefault(p => p.Sequence == sequence);
         }
 
+        internal void CreateParameterDefinition(Parameter parameter)
+        {
+            if (parameter == ThisParameter || parameter.Definition is not null)
+                return;
+            _owner.ParameterDefinitions.Add(new ParameterDefinition(parameter.Sequence, Utf8String.Empty, 0));
+        }
+
```

Comment:
The name `CreateParameterDefinition` suggests that it always creates a new parameter, however the implementation doesn't do this when the definition already exists. Should we throw instead? We could perhaps also introduce a `GetOrCreateDefinition` instead.